src/providers/polar.ts

```diff
@@ -0,0 +1,55 @@
+import { ofetch } from 'ofetch'
+import type { Provider, Sponsorship } from '../types'
+
+export const PolarProvider: Provider = {
+  name: 'polar',
+  fetchSponsors(config) {
+    return fetchPolarSponsors(config.polar?.token || config.token!)
+  },
+}
+
+export async function fetchPolarSponsors(token: string): Promise<Sponsorship[]> {
+  if (!token)
+    throw new Error('Polar token is required')
+
+  const apiFetch = ofetch.create({
+    baseURL: 'https://api.polar.sh/api/v1',
+    headers: { Authorization: `Bearer ${token}` },
+  })
+
+  /**
+   * First fetch the list of all subscriptions. This is a paginated
+   * endpoint so loop through all the pages
+   */
+  let page = 1
+  let pages = 1
+  const subscriptions = []
+  do {
+    const subs = await apiFetch('/subscriptions/subscriptions/search', { params: { page } })
+    subscriptions.push(...subs.items)
+
+    pages = subs.pagination.max_page
+    page += 1
+  } while (page <= pages)
+
+  return subscriptions
+    /**
+     * - People can subscribe for free on Polar : the price is null in this case
+     * - We also only keep `active` subscriptions. Still not sure what `inactive` means
```

Comment:
Active probably means past sponsors (monthly sponsors than canceled at some point)?

We do support listing past sponsors, where `monthlyDollars` should be `-1` for them and up to renderers to render them or not

Reply:
Yeah probably, that's what I thought too. But I'm not sure that the returned object will retains the same shape if `inactive`. That's why I suggested keeping the code as it is, until I or someone else can test the call API with past sponsorship, and adapt the code accordingly

But let's try it like that. I updated the code and and will keep an eye out as soon as anyone unsubscribes from my polar 👀